model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/stencils/apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence.py

```diff
@@ -60,14 +57,14 @@ def _apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence(
 
     z_nabla2_c = _calculate_nabla2_for_w(w_old, geofac_n2s)
 
-    w = where(
-        (interior_idx <= cell) & (cell < halo_idx),
+    w = concat_where(
+        (interior_idx <= CellDim) & (CellDim < halo_idx),
         _apply_nabla2_to_w(area, z_nabla2_c, geofac_n2s, w_old, diff_multfac_w),
         w_old,
     )
 
-    w = where(
-        (0 < k) & (k < nrdmax) & (interior_idx <= cell) & (cell < halo_idx),
+    w = concat_where(
+        (0 < KDim) & (KDim < nrdmax) & (interior_idx <= CellDim) & (CellDim < halo_idx),
```

Comment:
can we do this?
```suggestion
        (0 < KDim < nrdmax) & (interior_idx <= CellDim < halo_idx),
```

Reply:
This sadly doesn't work with embedded. Hopefully we can throw an error for it because it works for gtfn.